creator-node/src/blacklistManager.js

```diff
@@ -213,22 +243,44 @@ class BlacklistManager {
 
   /**
    * Retrieves all the deduped CIDs from the params and builds a mapping to <trackId: segments> for explicit trackIds (i.e. trackIds from table, not tracks belonging to users).
-   * @param {Object} param
-   * @param {number[]} param.allTrackIds all the trackIds to find CIDs for
-   * @param {Set<number>} param.explicitTrackIds explicit trackIds from table. Used to create mapping of <trackId: segments> in redis
-   * @returns
-   * {
-   *    segmentCIDs: <string[]> all CIDs that are BL'd
-   *    trackIdToSegments: {Object<number, string[]>} trackId : CIDs
-   * }
+   * @param {number[]} allTrackIds all the trackIds to find CIDs for (explictly blacklisted tracks and tracks from blacklisted users)
+   * @returns {string[]} all CIDs that are blacklisted from input track ids
    */
-  static async getCIDsToBlacklist({ allTrackIds, explicitTrackIds }) {
-    const tracks = await this.getAllCIDsFromTrackIdsInDb(allTrackIds)
+  static async getCIDsToBlacklist(inputTrackIds) {
+    const tracks = await this.getAllCIDsFromTrackIdsInDb(inputTrackIds)
 
     const segmentCIDs = new Set()
-    const trackIdToSegments = {}
 
-    return { segmentCIDs: [...segmentCIDs], trackIdToSegments }
+    // Retrieve CIDs from the track metadata and build mapping of <trackId: segments>
+    for (const track of tracks) {
+      if (!track.metadataJSON || !track.metadataJSON.track_segments) continue
+
+      for (const segment of track.metadataJSON.track_segments) {
+        if (!segment.multihash || CID_WHITELIST.has(segment.multihash)) continue
+
+        segmentCIDs.add(segment.multihash)
+      }
+    }
+
+    // also retrieves the CID's directly from the files table so we get copy320
+    if (inputTrackIds.length > 0) {
+      const files = await models.File.findAll({
+        where: {
+          trackBlockchainId: inputTrackIds
+        }
+      })
+      for (const file of files) {
+        if (
+          file.type === 'track' ||
+          file.type === 'copy320' ||
+          !CID_WHITELIST.has(file.multihash)
+        ) {
+          segmentCIDs.add(file.multihash)
+        }
+      }
+    }
+
+    return [...segmentCIDs]
```

Comment:
this already existed in the code base prior to this commit where this block was removed. https://github.com/AudiusProject/audius-protocol/pull/3250

the only difference is that we check the whitelist here too and remove any part of the code where we build the `trackIdToSegment` mapping